### ISSUE TITLE
allow using canonical transcript instead of highest impact

### ISSUE THREAD

**Issue by brentp** [arq5x/gemini#663]
see: https://groups.google.com/forum/#!topic/gemini-variation/KKCO05-RNYo

**Comment by roryk** [comment]
Hey ya'll,

What do you think we'd have to do to implement this? I've had some folks we work with asking about it.

**Comment by udp3f** [comment]
I like Brent's idea of '--use-canonical' flag for those who want ranking based on canonical transcripts. The default can still remain the same.

**Comment by roryk** [comment]
Is the canonical transcript flagged in the snpEFF/VEP output and folded into the INFO field somehow?

**Comment by udp3f** [comment]
Yes there's a '--canonical' flag in VEP and that should be available in INFO. Not sure about snpEff though.

**Comment by brentp** [comment]
as long as there is a way to know which is the Canonical (is it put first?) This should be implemented by adding a flag to gemini and then passing it to geneimpacts module which does the variant prioritization.

**Comment by jxchong** [comment]
++ particularly on being able to make hgvs use the canonical transcript

**Comment by davemcg** [comment]
Is there any kind of workaround we could implement before this feature gets added? Perhaps some way to get the VEP --pick (https://useast.ensembl.org/info/docs/tools/vep/script/vep_options.html) information added to variant_impact?

**Comment by brentp** [comment]
Hi @oleraj I'd gladly accept a PR for this.

**Comment by oleraj** [comment]
After doing some more digging, it looks like this was maybe already fixed by @roryk in geneimpacts (at least for VEP-annotated VCFs), though I haven't tested it yet. 
https://github.com/brentp/geneimpacts/commit/c1fd841c65a65e83be7fe1d1304785bb6db0642d#diff-ef46603b09e1d94334dfde203c2a72db
This is in at least version 0.3.4 for geneimpacts; can we update this in the requirements file for gemini?  Maybe geneimpacts>=0.3.6?  Currently it's using 0.1.3.

**Comment by jxchong** [comment]
@brentp Once geneimpacts is updated in the newest release, does that mean GEMINI will default to loading the canonical impact and not the most severe?  cc @oleraj

**Comment by brentp** [comment]
@jxchong. No. someone would have to implement an option (in gemini) to allow it to choose canonical transcripts).

**Comment by arq5x** [comment]
Hmm, I am not sure using the canonical impact is the best approach in the context of rare disease. I would rather manually refute candidates than miss them because they are on a different transcript.

**Comment by jxchong** [comment]
@arq5x Yes it's a mixed bag (why I asked). For discovery, I'd want the most severe impact, but for reporting out, I might prefer canonical transcript (it's not uncommon for us to sometimes pull out a severe impact in a incomplete/suspect/unsupported transcript and a synonymous change in all other transcripts)

**Comment by roryk** [comment]
Agreed, I don't really like using the canonical approach because you miss impactful variants. We missed a P53 variant because it was on a non-canonical transcript recently. But the canonical setting is useful when talking to other folks, I've found that clinicians know mutations by the amino acid change in the canonical transcript, so to be on the same page, we have to be talking about the same thing.

**Comment by brentp** [comment]
Just a reminder that the canonical transcript should be available in the variant_impacts table.
